feat: Added `GEOMETRY_STATUS_SHOW_CODE` and configuration.

alxbl opened the pull request:
As requested, I've ported this to the `mnml` branch. After looking at the `status` function, I figured that it made a lot of sense to have this as an option of status, so I added a configuration to show exit status.

If you'd prefer this to be a standalone function (to make it possible to have it in RPROMPT or INFO instead, let me know.

alxbl committed:
feat: Added `GEOMETRY_STATUS_SHOW_CODE` and configuration.

alxbl committed:
doc: Added documentation for status code display.

alxbl committed:
doc: Added documentation for status code display.

jedahan commented:
Thanks for porting! 

I think standalone is nicer, but I wonder if we should a) be exporting LAST_STATUS, or b) simply setting `?` to what it should be afterwards? If its not too much trouble to have it be a separate function like `geometry_exitcode` that would be awesome, otherwise I will just merge this.

jedahan commented on functions/geometry_status.zsh:
```diff
@@ -7,6 +7,9 @@ geometry_status() {
   : ${GEOMETRY_STATUS_SYMBOL_ROOT_ERROR:=▽}
+  : ${GEOMETRY_STATUS_SHOW_CODE:=false}
+  : ${GEOMETRY_STATUS_CODE_PREFIX:=' '}
+  : ${GEOMETRY_STATUS_CODE_SUFFIX:=' '}
```
how about using `GEOMETRY_SEPARATOR` instead of `PREFIX`/`SUFFIX`?

jedahan replied:
how about using `GEOMETRY_SEPARATOR` instead of `PREFIX`/`SUFFIX`?

jedahan replied:
how about using `GEOMETRY_SEPARATOR` instead of `PREFIX`/`SUFFIX`?

alxbl committed:
refactor: added exitcode function.

alxbl committed:
refactor: added exitcode function.

alxbl commented:
I pulled the `GEOMETRY_LAST_STATUS` into the precmd hook, which was necessary for it to be readable while building the RPROMPT. I also made exit code into its own function as you said. This way it can be put on either side, and will use `GEOMETRY_SEPARATOR`.

This should be the most modular approach. I'll update the doc in a few minutes.

Cheers,
Alex

alxbl committed:
doc: Added documentation for geometry_exitcode

alxbl committed:
doc: Added documentation for geometry_exitcode

alxbl commented:
One last thing, If you have an idea for a good symbol for the exit code (I was thinking unicode cross?), I can also add the symbol for it.

jedahan committed:
count lines with tokei

we need to test against previous lines

Commit:
Fix git symbol, thanks @Konsonanz

Commit:
Do not export fun variable when loading function, fixes #269

kyounger committed:
Refactor kube

Use better approach for retreiving current namespace that doesn't rely on
getting context first.

Use the geometry_kube_{namespace, symbol, context, version} convention to
make this function similar to the git function.

Detect a valid current context in addition to a non-zero KUBECONFIG.

Add GEOMETRY_KUBE_VERSION_COLOR.
Remove GEOMETRY_KUBE_VERSION check, as you can remove this element now
with by intentionally excluding geometry_kube_version from your prompt.

Added some different logic to pinning. If you have have GEOMETRY_KUBE_PIN
unset, then it defaults to false and displays prompt extras only if
valid KUBECONFIG or kubectl get current-context is valid. If explictly
set to false, then it will not display at all. If set to true, then it
displays all the time regardless of a valid context.

alxbl committed:
chore: merge with upstream/mnml

alxbl commented:
I tried to rebase this on the latest mnml but had a lot of trouble, so I ended up merging. Conflicts should be fixed. Feel free to squash merge.

jedahan commented:
Merged it in, thanks for the contribution!

jedahan commented:
see https://github.com/geometry-zsh/geometry/commit/0721dbbb5028f9b410df7422a3d37e3ef65dc9e0
